PXC-3508 : SST with OpenSSL 1.1.1 and encrypt=2,3 fails with "E SSL_C…

kennt-percona opened the pull request:
…TX_set_tmp_dh" from socat

Issue
Using OpenSSL fixed a bug with the way it used/generated dhparams.  This means that
certificates generated using an older version of OpenSSL would now fail if
used with OpenSSL 1.1.1.

Solution
Explicitly set the dhparam option with socat to workaround the use of the
old certs.

kamil-holubicki reviewed (commented):
Looks good, but do we have MTR test that tests it?

kennt-percona commented:
https://pxc.cd.percona.com/job/pxc56-test-mtr/60/
All nodes failing but due to:
galera_sst_allowed_methods (all nodes)
galera_sst_mysqldump_with_key (on buster, which I thinks is due to some SSL build issue with mysqldump)

kamil-holubicki reviewed (commented):
LGTM, but I have 1 question: shouldn't we use GCA for this? I see the same done for 5.7
https://github.com/percona/percona-xtradb-cluster/pull/1338

kennt-percona committed:
PXC-3508 : SST with OpenSSL 1.1.1 and encrypt=2,3 fails with "E SSL_CTX_set_tmp_dh" from socat

Issue
Using OpenSSL fixed a bug with the way it used/generated dhparams.  This means that
certificates generated using an older version of OpenSSL would now fail if
used with OpenSSL 1.1.1.

Solution
Explicitly set the dhparam option with socat to workaround the use of the
old certs.

kennt-percona commented:
@kamil-holubicki  Redid the 5.6 branch using the GCA process
